tests/src/system/basic/WskBasicNodeTests.scala

```diff
@@ -80,6 +81,101 @@ class WskBasicNodeTests
             }
     }
 
+    it should "Ensure that whisk.invoke() returns a promise" in withAssetCleaner(wskprops) {
+        val expectedDuration = 3 seconds
+
+        (wp, assetHelper) =>
+            val asyncName = "asyncAction"
+            val asyncAction = Some(TestUtils.getTestActionFilename("threeSecondRule.js"))
+
+            assetHelper.withCleaner(wsk.action, asyncName) {
+                (action, _) =>
+                    action.create(asyncName, asyncAction)
+            }
+
+            // this action does not supply a 'next' callback to whisk.invoke()
+            // and utilizes the returned promise
+            val invokeActionName = "invokeAction"
+            val invokeAction = Some(TestUtils.getTestActionFilename("invokePromise.js"))
+
+            assetHelper.withCleaner(wsk.action, invokeActionName) {
+                (action, _) =>
+                    action.create(invokeActionName, invokeAction)
+            }
+
+            var start = System.currentTimeMillis()
+            val runResolve = wsk.action.invoke(invokeActionName, Map("resolveOrReject" -> "resolve".toJson))
+            withActivation(wsk.activation, runResolve) {
+                activation =>
+                    activation.getFieldPath("response", "result", "activationId") shouldBe defined
+                    activation.getFieldPath("response", "result", "error") should not be defined
+                    activation.getFieldPath("response", "result", "result", "message") should be(Some {
+                        "Three second rule!".toJson
+                    })
+
+                    val duration = System.currentTimeMillis() - start
+                    duration should be >= expectedDuration.toMillis
+            }
+
+            start = System.currentTimeMillis()
+            val runReject = wsk.action.invoke(invokeActionName, Map("resolveOrReject" -> "reject".toJson))
+            withActivation(wsk.activation, runReject) {
+                activation =>
+                    activation.getFieldPath("response", "result", "activationId") should not be defined
+                    activation.getFieldPath("response", "result", "error") shouldBe defined
+
+                    val duration = System.currentTimeMillis() - start
+                    duration should be >= expectedDuration.toMillis
+            }
+    }
+
+    it should "Ensure that whisk.invoke() still uses a callback when provided one" in withAssetCleaner(wskprops) {
+        val expectedDuration = 3 seconds
+
+        (wp, assetHelper) =>
+            val asyncName = "asyncAction"
+            val asyncAction = Some(TestUtils.getTestActionFilename("threeSecondRule.js"))
```

Comment:
I may have added the following comment to threeSecondRule.js after you made this comment. Let me know if it is still unclear:

```javascript
/*
 * After 3 seconds, either resolve or reject depending on the value
 * of the resolveOrReject parameter.
 *
 * This craziness is so that the whisk.invoke() promise support can be
 * thoroughly tested (for invocations that both resolve and reject).
 */
```